src/js/components/Ballot/EmailBallotModal.jsx

```diff
@@ -124,23 +124,28 @@ export default class EmailBallotModal extends Component {
 
   ballotEmailSend () {
     let success_message = "";
+    let verification_email_sent = false;
     if ( this.state.on_facebook_login_step ) {
       success_message = <span>Success! This ballot has been sent to Facebook and the email address {this.state.sender_email_address} </span>;
     } else {
       success_message = <span>Success! This ballot has been sent to the email address {this.state.sender_email_address} </span>;
     }
 
     FriendActions.emailBallotData("", "", "", this.state.sender_email_address, this.state.email_ballot_message,
-      this.state.ballot_link, this.state.sender_email_address);
-    console.log("ballotEmailSend: email_address: ", this.state.sender_email_address);
+      this.state.ballot_link, this.state.sender_email_address, this.state.verification_email_sent);
+
+    if (!this.hasValidEmail()) {
```

Comment:
Not sure I understand the naming convention used here. Is this for the checking of a proper email address or an email address that has not been verified yet? I can't determine the condition of the check based on `hasValidEmail` being false. If it is false, to me the name does not state clearly what is not valid about it.

Reply:
I will discuss with @jainanisha90 a better function name. Thank you.